seed/challenges/01-front-end-development-certification/basic-javascript.json

```diff
@@ -1891,16 +1891,17 @@
       "releasedOn": "January 1, 2016",
       "head": [
         "var logOutput = \"\";",
-        "var oldLog = console.log;",
+        "var originalConsole = console",
         "function capture() {",
+        "    var oldLog = console.log;",
         "    console.log = function (message) {",
         "        logOutput = message + ''.trim();",
-        "        oldLog.apply(console, arguments);",
+        "        oldLog.apply(originalConsole, arguments);",
```

Comment:
Actually that's precisely the problem, if you notice I have brought the line 
`var oldLog = console.log;` inside the `capture` method.

By storing a instance of the `console.log` in `oldLog`, we are doing the `apply` call on different `caller` and `callee` objects. There are three scopes to consider here, `Global`, `capture` and our custom `console.log` 

Since this piece of code is executed inside a `eval` in non [strict mode](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Strict_mode), the scopes of the `console` object instances are going crazy when they come into the stack.